### PR TITLE
controller/deployer: Add logging.

### PR DESCRIPTION
**Pull request by archSeer**
Signed-off-by: Blaž Hrastnik <blaz.hrast@gmail.com>

### CODE REVIEW

File: controller/deployer/main.go [hunk 1]
```diff
@@ -18,25 +18,31 @@ import (
...
+	log    log15.Logger
 }
 
 func main() {
+	log := log15.New("app", "deployer")
+
 	client, err := controller.NewClient("", os.Getenv("AUTH_KEY"))
 	if err != nil {
-		log.Fatalln("Unable to create controller client:", err)
+		log.Error("Unable to create controller client", "err", err, "comment")
```
Comment (titanous) [inline controller/deployer/main.go]: extra `"comment"`

File: controller/deployer/main.go [hunk 2]
```diff
@@ -59,25 +66,26 @@ func main() {
 }
 
 func (c *context) HandleJob(job *que.Job) (e error) {
+	log := c.log.New("fn", "HandleJob")
```
Comment (titanous) [inline controller/deployer/main.go]: This should have all of the relevant IDs

**Comment by archSeer** [comment]
@titanous Comments addressed.

File: controller/deployer/main.go [hunk 3]
```diff
@@ -101,29 +117,37 @@ func (c *context) HandleJob(job *que.Job) (e error) {
...
 	events <- ct.DeploymentEvent{
 		ReleaseID: deployment.NewReleaseID,
 		Status:    "complete",
 	}
+	log.Info("Deployment complete", "at", "done")
 	return nil
 }
 
 func (c *context) rollback(deployment *ct.Deployment, original *ct.Formation) error {
+	log := c.log.New("fn", "rollback")
```
Comment (titanous) [inline controller/deployer/main.go]: This function should be passed the log context with all of the IDs

File: controller/deployer/strategies/all-at-once.go [hunk 4]
```diff
@@ -1,20 +1,26 @@
...
 
 import (
+	"github.com/flynn/flynn/Godeps/_workspace/src/gopkg.in/inconshreveable/log15.v2"
 	"github.com/flynn/flynn/controller/client"
 	ct "github.com/flynn/flynn/controller/types"
 )
 
-func allAtOnce(client *controller.Client, d *ct.Deployment, events chan<- ct.DeploymentEvent) error {
+func allAtOnce(l log15.Logger, client *controller.Client, d *ct.Deployment, events chan<- ct.DeploymentEvent) error {
+	l.Info("fn", "allAtOnce")
```
Comment (titanous) [inline controller/deployer/strategies/all-at-once.go]: First arg should be message? Do it after assigning log maybe?

File: controller/deployer/strategies/one-by-one.go [hunk 5]
```diff
@@ -1,20 +1,26 @@
...
 
 import (
+	"github.com/flynn/flynn/Godeps/_workspace/src/gopkg.in/inconshreveable/log15.v2"
 	"github.com/flynn/flynn/controller/client"
 	ct "github.com/flynn/flynn/controller/types"
 )
 
-func oneByOne(client *controller.Client, d *ct.Deployment, events chan<- ct.DeploymentEvent) error {
+func oneByOne(l log15.Logger, client *controller.Client, d *ct.Deployment, events chan<- ct.DeploymentEvent) error {
+	l.Info("fn", "oneByOne")
```
Comment (titanous) [inline controller/deployer/strategies/one-by-one.go]: Ditto

**Comment by titanous** [comment]
LGTM

**Pull request merged**